### PR TITLE
Fix deadlock between PDP and StatefulReader

### PR DESCRIPTION
**Pull request by sloretz**
This is intended to fix a deadlock that causes tests to hang on the ROS 2 buildfarm (ros2/build_cop#248) using Fast-RTPS `v1.9.2`. There's more info here: https://github.com/ros2/build_cop/issues/248#issuecomment-555625379.

This PR avoids deadlock by releasing the PDP mutex before calling `remove_remote_participant()`. It seems like a reasonable thing to do since `remove_remote_participant()` [acquires the lock internally](https://github.com/eProsima/Fast-RTPS/blob/daabbc72b79b094cb9fe684655726105f27bdc7d/src/cpp/rtps/builtin/discovery/participant/PDP.cpp#L856), though I don't know if releasing the lock momentarily breaks any assumptions in `check_remote_participant_lifeliness()`.

### CODE REVIEW

**Comment by sloretz** [comment]
@richiprosima would it be possible to get this reviewed and merged into the 1.9.x branch this week?


@dirk-thomas @mjcarroll FYI

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/963/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/801/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/960/)
* Performance [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Performance%20Tests%20Localhost/281/)

File: src/cpp/rtps/builtin/discovery/participant/PDP.cpp [hunk 1]
```diff
@@ -1028,7 +1028,9 @@ void PDP::check_remote_participant_liveliness(
                 std::chrono::microseconds(TimeConv::Duration_t2MicroSecondsInt64(remote_participant->m_leaseDuration));
         if (now > real_lease_tm)
         {
+            guard.unlock();
             remove_remote_participant(remote_participant->m_guid, ParticipantDiscoveryInfo::DROPPED_PARTICIPANT);
+            guard.lock();
```
Comment (richiware) [inline src/cpp/rtps/builtin/discovery/participant/PDP.cpp]: Not necessary to take again the mutex because the function will return immediately.

Comment (dirk-thomas) [reply src/cpp/rtps/builtin/discovery/participant/PDP.cpp]: @richiware can you do the necessary edit through the web interface?

Comment (richiware) [reply src/cpp/rtps/builtin/discovery/participant/PDP.cpp]: Done

**Comment by richiware** [comment]
After running the CI I will merge the PR and tomorrow I will make the backport to 1.9.x

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/978/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/811/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/974/)
* Performance [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Performance%20Tests%20Localhost/289/)

**Comment by richiware** [comment]
Failed test not related.

**Pull request merged**